invenio_vocabularies/contrib/awards/config.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Vocabulary awards configuration."""
+
+from flask import current_app
+from flask_babelex import lazy_gettext as _
+from invenio_records_resources.services import SearchOptions
+from invenio_records_resources.services.records.components import DataComponent
+from invenio_records_resources.services.records.params import \
+    SuggestQueryParser
+from werkzeug.local import LocalProxy
+
+from ...services.components import ModelPIDComponent
+
+award_schemes = LocalProxy(
+    lambda: current_app.config["VOCABULARIES_AWARD_SCHEMES"]
+)
+
+
+class AwardsSearchOptions(SearchOptions):
+    """Search options."""
+
+    suggest_parser_cls = SuggestQueryParser.factory(
+        fields=[
+            'title.*^50',
+            'title.*._2gram',
+            'title.*._3gram',
+            'number^10'
+        ],
+    )
```

Comment:
It could, but I feel with a facet it's a bit more "streamlined", i.e. no need to string-format to build a query and potentially have to escape characters or whatever... Could also be just mental leftovers from Zenodo :)

Reply:
you are right, we would need to reinvent the wheel for formatting. I only had the filtering case in mind.

Reply:
@slint This PR is ready, faceting configuration happens at RDM-Records level (see [example](https://github.com/inveniosoftware/invenio-rdm-records/blob/aaf8df982e9099b12b38d3374519669c9ca24f68/invenio_rdm_records/services/facets.py#L46)) and I think we can do with the `VocabularyLabels`, which will show the i18n title of the funder/award. And are configured [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/cc8514638bbd052ae211772c8ca542fa01f5c4ec/invenio_rdm_records/config.py#L308)... I'm not 100% sure this fulfills our use case though.

What we want is a facet over the funder when querying awards, right? I think in RDM Records all faceting is configured **only** for the main search. So we might need to do some extra work. Treat in a different PR?

Reply:
This is actually for filtering over `/api/awards`, and more specifically for the suggester (though I think it wouldn't matter with the way we build queries?). Imagine the following:

User selects the funder "European Commision" (with `pid = 00k4n6c32`) from the modal's dropdown. Afterwards, there's the first query on `/api/awards?funder=00k4n6c32&suggest=` (no search query was typed by the user), which returns only awards from the selected funder. Then if the user types e.g. an acronym in the search box (e.g. `Si-POP`), we do the query `/api/awards?funder=00k4n6c32&suggest=Si-POP`.

Then, there are things to take into account on how we search published records, but can be discussed separately. One example is "I want all records funded by the EC", which translates to something like `/api/records?q=funding.funder.id:00k4n6c32`.

Reply:
Okay, I got it now (thanks for the detailed explanation), it would be done in vocabularies then. However, since accounts got fixed... maybe we can start merging and treating in separate prs so manuel does not need to install from many masters 😅